src/Http/Controllers/SearchController.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Rapidez\Core\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+
+class SearchController
+{
+    public function __invoke(Request $request)
+    {
+        $query = Str::lower($request->q);
+        $searchQueryModel = config('rapidez.models.search_query');
+        if ($searchQuery = $searchQueryModel::firstOrCreate(
+            ['query_text' => $query],
+            ['store_id' => config('rapidez.store')]
+        )) {
+            $searchQuery->increment('popularity');
```

Comment:
So when the query doesn't exists we're first inserting it and then we update it?